core/src/main/java/com/linecorp/armeria/common/outlier/OutlierDetectionBuilder.java

```diff
@@ -0,0 +1,139 @@
+            OutlierDetection.builder(OutlierDetectingRule.of())
+                            .build();
+
+    private final OutlierDetectingRule rule;
+
+    private Ticker ticker = Ticker.systemTicker();
+    private Duration counterSlidingWindow = Duration.ofSeconds(DEFAULT_COUNTER_SLIDING_WINDOW_SECONDS);
+    private Duration counterUpdateInterval = Duration.ofSeconds(DEFAULT_COUNTER_UPDATE_INTERVAL_SECONDS);
+    private double failureRateThreshold = DEFAULT_FAILURE_RATE_THRESHOLD;
+    private long minimumRequestThreshold = DEFAULT_MINIMUM_REQUEST_THRESHOLD;
+
+    OutlierDetectionBuilder(OutlierDetectingRule rule) {
+        requireNonNull(rule, "rule");
+        this.rule = rule;
+    }
+
+    /**
+     * Sets the {@link Ticker} to use for measuring elapsed time.
+     */
+    public OutlierDetectionBuilder ticker(Ticker ticker) {
+        requireNonNull(ticker, "ticker");
+        this.ticker = ticker;
+        return this;
+    }
+
+    /**
+     * Sets the time length of sliding window to accumulate the count of events.
+     * Defaults to {@value #DEFAULT_COUNTER_SLIDING_WINDOW_SECONDS} seconds if unspecified.
+     */
+    public OutlierDetectionBuilder counterSlidingWindow(Duration counterSlidingWindow) {
+        requireNonNull(counterSlidingWindow, "counterSlidingWindow");
+        checkArgument(!counterSlidingWindow.isNegative() && !counterSlidingWindow.isZero(),
+                      "counterSlidingWindow: %s (expected: > 0)", counterSlidingWindow);
+        this.counterSlidingWindow = counterSlidingWindow;
+        return this;
+    }
+
+    /**
+     * Sets the time length of sliding window to accumulate the count of events, in milliseconds.
+     * Defaults to {@value #DEFAULT_COUNTER_SLIDING_WINDOW_SECONDS} seconds if unspecified.
+     */
+    public OutlierDetectionBuilder counterSlidingWindowMillis(long counterSlidingWindowMillis) {
+        return counterSlidingWindow(Duration.ofMillis(counterSlidingWindowMillis));
+    }
+
+    /**
+     * Sets the interval that an {@link OutlierDetector} can see the latest accumulated count of events.
+     * Defaults to {@value #DEFAULT_COUNTER_UPDATE_INTERVAL_SECONDS} second if unspecified.
+     */
+    public OutlierDetectionBuilder counterUpdateInterval(Duration counterUpdateInterval) {
+        requireNonNull(counterUpdateInterval, "counterUpdateInterval");
+        checkArgument(!counterUpdateInterval.isNegative() && !counterUpdateInterval.isZero(),
+                      "counterUpdateInterval: %s (expected: > 0)", counterUpdateInterval);
+        this.counterUpdateInterval = counterUpdateInterval;
+        return this;
+    }
+
+    /**
+     * Sets the interval that an {@link OutlierDetector} can see the latest accumulated count of events,
+     * in milliseconds. Defaults to {@value #DEFAULT_COUNTER_UPDATE_INTERVAL_SECONDS} second if unspecified.
+     */
+    public OutlierDetectionBuilder counterUpdateIntervalMillis(long counterUpdateIntervalMillis) {
```

Comment:
nit; is there no need to check if the value is positive?

Reply:
Ditto.